File: python/queue_utils.py

from collections import deque

class QueueUtils:

    def is_palindrome(self, text: str) -> bool:
       
        normalized_text = text.lower()

        original_queue = deque()
        
        stack = []

        for char in normalized_text:
            original_queue.append(char) 
            stack.append(char)          

        inverted_queue = deque()

        while stack:
            inverted_queue.append(stack.pop())

        while original_queue:
            if original_queue.popleft() != inverted_queue.popleft():
                return False

        return True

File: src/utils/QueueUtils.java
package utils;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Stack;

public class QueueUtils {

    public boolean isPalindrome(String str) {
        
        String normalizedStr = str.toLowerCase();

        Queue<Character> originalQueue = new LinkedList<>();
        Stack<Character> stack = new Stack<>();

        for (char c : normalizedStr.toCharArray()) {
            originalQueue.offer(c); // Cola: Orden natural
            stack.push(c); // Pila: Orden inverso para luego llenar la segunda cola
        }

        Queue<Character> invertedQueue = new LinkedList<>();

        while (!stack.isEmpty()) {
            invertedQueue.offer(stack.pop());
        }

        while (!originalQueue.isEmpty()) {
           
            if (!originalQueue.poll().equals(invertedQueue.poll())) {
                return false;
            }
        }

        return true;
    }
}